os/pm/pm_timedsuspend.c

```diff
@@ -112,18 +128,17 @@ static void timer_timeout(int argc, int domain_id, uint32_t hash_pid)
 
 int pm_timedsuspend(int domain_id, unsigned int milliseconds)
 {
-	int hash_pid = PIDHASH(getpid());
+	int pid = getpid();
 
 	if (domain_id >= CONFIG_PM_NDOMAINS || pm_domain_map[domain_id]==NULL) {
 		set_errno(EINVAL);
 		return ERROR;
 	}
 
 	/* Check if there is already a wdog lock timer running for 
-	 * the process */
-	if (pid_timer_map[hash_pid] != NULL) {
-		pmdbg("There is already a lock timer running for this process pid %d with domain %s\n", getpid(), pm_domain_map[domain_id]);
-		return ERROR;
+	 * the process, if so then cancel it */
+	if (pid_timer_map[pid] != NULL) {
```

Comment:
I dont think this check is necessary here. You are already checking it inside the function.